File: Programming Fundamentals/C#/Fundamentals/03.Arrays/Lab/P07_EqualArrays/P07_EqualArrays.cs
namespace P07_EqualArrays
{
    using System;
    using System.Linq;

    class P07_EqualArrays
    {
        static void Main(string[] args)
        {
            int[] arr1 = Console.ReadLine()
                                .Split()
                                .Select(int.Parse)
                                .ToArray();
            int[] arr2 = Console.ReadLine()
                                .Split()
                                .Select(int.Parse)
                                .ToArray();

            if (arr1.Length != arr2.Length)
            {
                Console.WriteLine("Arrays are not identical.");
            }

            int sum = 0;
            for (int i = 0; i < arr1.Length; i++)
            {
                if (arr1[i] != arr2[i])
                {
                    Console.WriteLine($"Arrays are not identical. Found difference at {i} index");
                    return;
                }
                else
                {
                    sum += arr1[i];
                }
            }

            Console.WriteLine($"Arrays are identical. Sum: {sum}");
        }
    }
}

File: Programming Fundamentals/JS/03.Arrays/Lab/p07_EqualArrays.js
function solve(arr1, arr2) {
  let sum = 0;
  for (let i in arr1) {
      if (arr1[i] !== arr2[i]) {
          console.log(`Arrays are not identical. Found difference at ${i} index`);
          return;
      }
      sum += arr1[i] - 0; //ASCII hack
  }

  console.log(`Arrays are identical. Sum: ${sum}`);
}

// Don't copy the calling of the function in judge, you won't get any points, just the code above
solve(['10', '20', '30'], ['10', '20', '30']);
solve(['1', '2', '3', '4', '5'], ['1', '2', '4', '4', '5']);
solve(['1'], ['10']);
